core/field_angle.js

```diff
@@ -209,7 +210,7 @@ Blockly.FieldAngle.prototype.showEditor_ = function() {
 
 
   Blockly.DropDownDiv.setColour(this.sourceBlock_.getColour(),
-      this.sourceBlock_.getColour());
+     goog.color.rgbArrayToHex(goog.color.lighten(goog.color.hexToRgb(this.sourceBlock_.getColour()), 0.6)));
```

Comment:
Bonus, BeksOmega just added an API for this. You can now use block.getColourBorder().

The one caveat is you need to check which border colour to use since the old style is slightly different. We might tweak the API with this use case in mind, but for now this can be:

var border = this.sourceBlock_.getColourBorder();
border = border.colourBorder == null ? border.colourLight : border.colourBorder;

Blockly.DropDownDiv.setColour(this.sourceBlock_.getColour(), border);